pybossa/util.py

```diff
@@ -1051,20 +1051,27 @@ def get_unique_user_preferences(user_prefs):
     return duser_prefs
 
 
-def get_user_pref_db_clause(user_pref):
+def get_user_pref_db_clause(user_pref, user_email):
     # expand user preferences as per sql format for jsonb datatype
     # single user preference with multiple value or
     # multiple user preferences with single/multiple values
     _valid = ((k, v) for k, v in user_pref.iteritems() if isinstance(v, list))
     user_prefs = [{k: [item]} for k, pref_list in _valid
                   for item in pref_list]
-
+    assign_key = 'assign_user'
     if not user_prefs:
-        return 'task.user_pref IS NULL OR task.user_pref = \'{}\''
-
-    sql_strings = ('task.user_pref @> \'{}\''.format(json.dumps(up).lower())
+        user_pref_sql = '''(task.user_pref IS NULL OR task.user_pref = \'{}\' )'''
+        email_sql = ''' OR (task.user_pref->\'{}\' IS NOT NULL
+                AND task.user_pref @> \'{}\')
+                '''.format(assign_key, json.dumps({assign_key: [user_email]}).lower())
```

Comment:
Using format in SQL queries is dangerous - what are the possible values of `user_email`? I tested yesterday and it looks like the api allows users to enter any string as their email address. We should use query parameters or properly escape/sanitize the user input

Reply:
Yes, thats what she is going to do, this is not the latest.